lib/events.js

```diff
@@ -286,36 +290,36 @@ EventEmitter.prototype.prependOnceListener =
 // Emits a 'removeListener' event if and only if the listener was removed.
 EventEmitter.prototype.removeListener =
     function removeListener(type, listener) {
-      var list, events, position, i, originalListener;
-
       if (typeof listener !== 'function') {
         const errors = lazyErrors();
         throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'listener',
                                    'Function');
       }
 
-      events = this._events;
+      const events = this._events;
       if (events === undefined)
         return this;
 
-      list = events[type];
+      const list = events[type];
       if (list === undefined)
         return this;
 
       if (list === listener || list.listener === listener) {
-        if (--this._eventsCount === 0)
-          this._events = Object.create(null);
-        else {
-          delete events[type];
-          if (events.removeListener)
-            this.emit('removeListener', type, list.listener || listener);
-        }
-      } else if (typeof list !== 'function') {
-        position = -1;
+        events[type] = undefined;
+        --this._eventsCount;
```

Comment:
Having the `--` at the start here just looks odd. Having it at the end may be more readable

Reply:
Yeah, I'm not sure which way to go here. It matches the rest of the `events`. Feels more confusing if we go back and forth throughout this code. Maybe just convert all the usage witin this file (where the result isn't being used)?

Reply:
I'm okay with having consistency in this file.

It probably looks a lot less weird to you if you're used to writing C++ code :)